provider/azure/instance.go

```diff
@@ -139,7 +139,7 @@ func instancePublicIPAddresses(
 		return nil, nil
 	}
 	instancePips := make(map[instance.Id][]network.PublicIPAddress)
-	for ; pipsResult.NotDone(); err = pipsResult.NextWithContext(sdkCtx) {
+	for ; pipsResult.NotDone(); pipsResult.NextWithContext(sdkCtx) {
```

Comment:
I think we need to check this error as it appears that NextWithContext can do paging (as in make paginated API calls)